container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+// if toFormat is set to True, adds a . in front of the value so that the string can be formatted.
+func keyValueToMap(value []string) map[string]string {
+	convMap := make(map[string]string)
+	var temp []string
+	for _, kvpair := range value {
+		temp = strings.Split(kvpair, "=")
+		key, val := temp[0], temp[1]
+		convMap[key] = val
+	}
+	return convMap
+}
+
+type formattedString map[string]interface{}
+
+func formatWithMap(format string, params formattedString) string {
+	msg := &bytes.Buffer{}
+	template.Must(template.New("").Parse(format)).Execute(msg, params)
+	return msg.String()
+}
+
+// stamp provides the substitutions of variables inside {} using info in file pointed to
+// by stampInfoFile.
+func stamp(inp string) (string, error) {
+	if len(stampInfoFile) == 0 || inp == "" {
+		return inp, nil
+	}
+	formatArgs := make(map[string]interface{})
+	for _, infofile := range stampInfoFile {
+		f, err := os.Open(infofile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to open file %s", infofile)
+		}
+		defer f.Close()
+		// scanner reads line by line and discards '\n' character already
+		scanner := bufio.NewScanner(f)
+		var temp []string
+		for scanner.Scan() {
+			temp = strings.Split(scanner.Text(), " ")
+			key, val := temp[0], temp[1]
+			if _, ok := formatArgs[key]; ok {
+				fmt.Printf("WARNING: Duplicate value for key %s: using %s", key, val)
+			}
+			formatArgs[key] = val
+		}
+		if err = scanner.Err(); err != nil {
+			return "", errors.Wrapf(err, "failed to read line from file %s", infofile)
+		}
+	}
+	// do string manipulation in order to mimic python string format.
+	// specifically, replace '{' with '{{.' and '}' with '}}'.
+	inpReformatted := strings.ReplaceAll(inp, "{", "{{.")
+	inpReformatted = strings.ReplaceAll(inpReformatted, "}", "}}")
+	return formatWithMap(inpReformatted, formattedString(formatArgs)), nil
+}
+
+// mapToKeyValue reverses a map to a '='-separated array of strings in {key}={value} format
+func mapToKeyValue(kvMap map[string]string) []string {
+	keyVals := []string{}
+	concatenated := ""
+	for k, v := range kvMap {
+		concatenated = k + "=" + v
+		keyVals = append(keyVals, concatenated)
+	}
+	return keyVals
+}
+
+// resolveVariables resolves the environment variables embedded in the given value using
+// provided map of environment variable expansions.
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
+		case string:
+			return environment[p]
+		}
+		errorMet = true
+		return ""
+	}
+	if errorMet {
+		err := errors.New("environment variable sought after does not exist in evnrionemnet map")
+		if err != nil {
+			errors.Wrap(err, "failed to create new error")
+		}
+		return "", err
+	}
+	return os.Expand(value, mapper), nil
+}
+
+func main() {
```

Comment:
This function is too long. Suggest refactoring into functions.

Reply:
yep will refactor into a package under compat